### DISCUSSION TITLE
Linux Client

### DISCUSSION THREAD

**Discussion by Smcgorry** [cmu-sei/GHOSTS#231] [Q&A] [answered]
I'm having an issue when running the client on Linux. I'm using "Dotnet" to run the application, but when it boots up it freezes. I wanted to know if there is a solution to this issue or if I'm putting the wrong information into the json files in the Ghosts folder. I'm just trying to run the base application with no changes to timeline only changes to the application.json file linking it to the Server. 
![Capture](https://github.com/cmu-sei/GHOSTS/assets/137307976/2cb99acf-3d05-46f9-92e7-25e63ecefe83)

**Comment by sei-dupdyke** [comment]
Is there anything in the logs/application.log file please?

**Reply by Smcgorry** [reply]
I know where it is on the windows machine but for the Linux machine, I can't find the logs folder or application.log. I can't find where the error is occurring. If there's another file or folder, I can check let me know. 
![Capture](https://github.com/cmu-sei/GHOSTS/assets/137307976/b7257006-aaba-445f-bfd4-083a9ac3bdc9)

**Reply by sei-dupdyke** [reply]
Is there anything else in the terminal window to go on? Seems odd there are no logs created, and nothing in the console to indicate failure. Sorry, but I have not seen this happen before.

**Reply by Smcgorry** [reply]
Nothing else is created in the terminal window on the client side. When its run nothing is produced and essentially freezes up. No error messages are generated nor stored.

**Reply by sei-dupdyke** [reply]
Can you run it in debug mode as $ `dotnet ghosts.client.linux.dll --debug` please?

**Reply by Smcgorry** [reply]
![Capture](https://github.com/cmu-sei/GHOSTS/assets/137307976/260ddf86-55f3-4e8a-93bf-4fc784959e2e)
Sorry for the delay. Been busy with work. Here is what the command looks like run. I also wanted to add if there's any other command I can run to try and get the linux machine running. Or if you haven't built out the code to support a linux machine.

**Reply by sei-dupdyke** [reply]
Hmm, this seems to indicate it is working (at least up until this point).  What linux distro are you using please?

**Reply by Smcgorry** [reply]
I'm using Ubuntu and made some changes to the timeline that could potentially be causing issues. I shaped the timeline to fit a windows machine thinking the code would still be able to run it. If the timeline structure is different than a windows machine, can you, please sent the structure for it or at least an example?

**Reply by sei-dupdyke** [reply]
The formats are exactly the same, but that makes me wonder if there is an issue with something in the configuration. What else did you change in ./config?

If you create a ./logs folder - does it spit out any logging?

**Reply by Smcgorry** [reply]
I made a .log text file but nothing loads in the file so im assuming its not uploading the data which is odd. Is there anything I can check or change?
![image](https://github.com/cmu-sei/GHOSTS/assets/137307976/13a9dc03-c652-4120-90d7-84353d25f66d)

**Reply by sei-dupdyke** [reply]
No sorry, check the logs folder up above that file, what's in there?

**Comment by sei-dupdyke** [comment, marked as answer]
What handlers are in your timeline? If its Firefox or Chrome, do you have the corresponding chrome or geckodriver binaries?
